### PR TITLE
Polynomial modular addition research implementation and testing

### PR DESCRIPTION
**Pull request by feltroidprime**
related to #7 
lacks full Cairo translation of the Cython code
and edge case handling a=b=P, a+b=2P 
except that algo is correct

automatic mac support in makefile for cython code and parallelisation thanks to @0xflo-tez

### CODE REVIEW

File: tools/make/setup.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    file_name.append(e.sources[0].split('/')[1].split('.py')[0])
+print(folder_name)
+
+setup(
+    ext_modules=cythonize(extensions,
+                          compiler_directives={'language_level': "3", "binding": True},
+                          force=True, annotate=True, build_dir="build/cython"),
+    include_dirs=[numpy.get_include()])  # , root_path+'tools/c_algorithms/c-algorithms-1.2.0/'])
+
+# print(folder_name)
```
Comment (abdelhamidbakhta) [inline tools/make/setup.py:62]: remove commented code

File: tools/py/add_mod_poly_c.pyx [hunk 2]
```diff
@@ -0,0 +1,128 @@
+from libc.stdio cimport printf
```
Comment (abdelhamidbakhta) [inline tools/py/add_mod_poly_c.pyx:1]: can't we factorize code with the pure python code ?
Or at least separate in another directory

**Review by abdelhamidbakhta — APPROVED** [review]
left some comments but otherwise looks good to me

File: tools/py/add_mod_poly_c.pyx (hunk 2 above)
Comment (feltroidprime) [reply tools/py/add_mod_poly_c.pyx:1]: in Cython you select the functions that can be called from python (def ou cpdef functions)
here test_full_mod is the "allowed" python function (even if it calls cdef functions that are pure C and have no call overhead)
test_full_mod function is imported in the pure py in /https://github.com/keep-starknet-strange/garaga/blob/b02bb7dacf677939a08e0ff99cbdf39180a56dd8/tests/python_tests/add_mod_poly_extensive_test.py
overall .pyx files, after being compiled, acts (if you define them so with cpdef or def) as C-compiled libraries that can directly be imported

File: tools/make/setup.py (hunk 1 above)
Comment (feltroidprime) [reply tools/make/setup.py:62]: ok

File: tools/py/add_mod_poly_c.pyx (hunk 2 above)
Comment (feltroidprime) [reply tools/py/add_mod_poly_c.pyx:1]: However, add_mod_poly.py will be deleted and replaced by this .pyx file in the future for cleaner tools/py folder

**Pull request merged**